lib/foreman_inventory_upload/generators/fact_helpers.rb

```diff
@@ -57,55 +57,187 @@ def cloud_provider(host)
       end
 
       def obfuscate_hostname?(host)
+        # Returns true if hostname obfuscation should be applied for a given host.
+        # This method determines whether the hostname associated with a host
+        # should be hidden or masked based on global settings and host-specific facts.
+
+        # Check the global setting for hostname obfuscation.
+        # If 'obfuscate_inventory_hostnames' is set to true in the application settings,
+        # then hostname obfuscation is always enabled, regardless of host-specific settings.
+        return true if Setting[:obfuscate_inventory_hostnames]
+
+        # Retrieve the host-specific fact related to hostname obfuscation from 'insights_client'.
+        # This fact indicates whether hostname obfuscation is enabled for this particular host.
         insights_client_setting = fact_value(host, 'insights_client::obfuscate_hostname_enabled')
+
+        # Cast the retrieved setting to a boolean value.
+        # This ensures that the value from the fact (which might be a string or other type)
+        # is correctly interpreted as true or false.
         insights_client_setting = ActiveModel::Type::Boolean.new.cast(insights_client_setting)
+
+        # If the host-specific setting is not nil (meaning it was explicitly set to true or false),
+        # return its boolean value directly. This prioritizes the host-specific setting
+        # over the global default if the host has a specific configuration.
         return insights_client_setting unless insights_client_setting.nil?
 
+        # If the host-specific setting was nil (meaning it wasn't explicitly set for this host),
+        # fall back to the global 'obfuscate_inventory_hostnames' setting.
+        # This serves as the default behavior when no specific host configuration is found.
         Setting[:obfuscate_inventory_hostnames]
```

Comment:
Could be a bit more concise here with a ternary. But if you think it adds value I am also okay with leaving it more explicit, up to you.

```suggestion
        # If the host-specific setting is nil (meaning it wasn't explicitly set for this host),
        # fall back to the global 'obfuscate_inventory_hostnames' setting.
        # This serves as the default behavior when no specific host configuration is found.
        # Otherwise, return the host-specific setting directly.
        insights_client_setting.nil? ? Setting[:obfuscate_inventory_hostnames] : insights_client_setting
```